Userland/Libraries/LibWeb/Painting/PaintableBox.cpp

```diff
@@ -358,11 +366,16 @@ void PaintableBox::apply_clip_overflow_rect(PaintContext& context, PaintPhase ph
 
     if (!m_clipping_overflow) {
         context.painter().save();
-        context.painter().add_clip_rect(context.enclosing_device_rect(*clip_rect).to_type<int>());
+        auto device_clip_rect = context.enclosing_device_rect(*clip_rect).to_type<int>();
+        if (overflow_x != CSS::Overflow::Visible)
+            context.painter().add_horizontal_clip_rect(device_clip_rect);
+        if (overflow_y != CSS::Overflow::Visible)
+            context.painter().add_vertical_clip_rect(device_clip_rect);
         m_clipping_overflow = true;
     }
 
-    if (!clip_rect->is_empty() && overflow_y == CSS::Overflow::Hidden && overflow_x == CSS::Overflow::Hidden) {
+    // FIXME: Implement https://drafts.csswg.org/css-overflow/#corner-clipping properly
```

Comment:
This FIXME is a little vague. What is the issue if just `overflow-x` or just `overflow-y` is used?

Reply:
The spec says there are different ways to handle clipping corners when given different combinations of overflow-x and overflow-y.

I've changed the comment to be a bit more informative :)